sepal_ui/sepalwidgets/app.py

```diff
@@ -600,3 +643,62 @@ def _on_locale_select(self, change):
         su.set_config_locale(loc.code)
 
         return
+
+
+class ThemeSelect(v.Btn, SepalWidget):
+    """
+    A theme selector for sepal-ui based application.
+
+    It displays the currently requested theme (default to dark).
+    When value is changed, the sepal-ui config file is updated. It is designed to be used in a AppBar component.
+
+    .. versionadded:: 2.7.0
+
+    Args:
+        kwargs (dict, optional): any arguments for a Btn object, children and v_model will be override
+    """
+
+    THEME_ICONS = {"dark": "fas fa-moon", "light": "fas fa-sun"}
+    "dict: the dictionnry of icons to use for each theme (used as keys)"
+
+    theme = "dark"
+    "str: the current theme of the widget (default to dark)"
+
+    def __init__(self, **kwargs):
+
+        # get the current theme name
+        self.theme = sepal_ui.get_theme(sepal_ui.config_file)
+
+        # set the btn parameters
+        kwargs["x_small"] = kwargs.pop("x_small", True)
+        kwargs["fab"] = kwargs.pop("fab", True)
+        kwargs["class_"] = kwargs.pop("class_", "ml-2")
+        kwargs["children"] = [v.Icon(children=[self.THEME_ICONS[self.theme]])]
+        kwargs["v_model"] = self.theme
+
+        # create the btn
+        super().__init__(**kwargs)
+
+        # add some js events
+        self.on_event("click", self.toggle_theme)
+
+    def toggle_theme(self, widget, event, data):
+        """
+        toggle the btn icon from dark to light and adapt the configuration file at the same time
+        """
+        # use a cycle to go through the themes
+        theme_cycle = cycle(self.THEME_ICONS.keys())
+        next(t for t in theme_cycle if t == self.theme)
+        self.theme = next(t for t in theme_cycle)
+
+        # change icon
+        self.color = "info"
```

Comment:
Can we also change the button color? depending on the current theme?

Reply:
sure what do you have in mind (colors are proposition I'm pretty sure there will be modifications) ?

Reply:
Well, let's leave this conversation as it is, and later we can check when agreeing with a new light theme proposition.